crates/ln-dlc-node/src/blockchain.rs

```diff
@@ -0,0 +1,104 @@
+use crate::bitcoin_conversion::to_tx_30;
+use crate::node::Storage;
+use anyhow::Context;
+use anyhow::Result;
+use bdk_esplora::esplora_client;
+use bitcoin::consensus::encode::serialize_hex;
+use bitcoin::Block;
+use bitcoin::BlockHash;
+use bitcoin::Transaction;
+use bitcoin::Txid;
+use std::sync::Arc;
+use tracing::instrument;
+
+const SOCKET_TIMEOUT: u64 = 30;
+
+#[derive(Clone)]
+pub struct Blockchain<N> {
+    // TODO: We should aim to only have an async client.
```

Comment:
❓ Why don't we use the async esplora client already? Too much effort to do it right away?

Reply:
Tbh, I'm not even sure if the TODO is valid.

We _do_ use the async client, but not in all cases. We use it for the on-chain sync, which is an `async` function.

We use the blocking client in other spots where we are in a blocking context: primarily `rust-dlc` trait implementations.

Reply:
I've tried to clarify this with documentation.

I also added an async `broadcast_transaction` method, since there are a couple places were it might be better than the blocking version.